### PR TITLE
feat: add tests coverage

### PR DESCRIPTION
**Pull request by bolinocroustibat**
Closes https://github.com/datagouv/hydra/issues/121.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,6 +69,11 @@ Then you can run the tests with `poetry run pytest`.
 
 If you would like to see print statements as they are executed, you can pass the -s flag to pytest (`poetry run pytest -s`). However, note that this can sometimes be difficult to parse.
 
+### Tests coverage
+
+Pytest automatically uses the `coverage` package to generate a coverage report, which is displayed at the end of the test run in the terminal.
+The converage is configured in the `pypoject.toml` file, in the `[tool.pytest.ini_options]` section.
```
Comment (magopian) [inline README.md:75]: Typo: `coverage`

File: README.md [hunk 2]
```diff
@@ -69,6 +69,11 @@ Then you can run the tests with `poetry run pytest`.
...
+You can also overrides the coverage report configuration when running the tests by passing some flags like `--cov-report` to pytest. See [the pytest-cov documentation](https://pytest-cov.readthedocs.io/en/latest/config.html) for more information.
```
Comment (magopian) [inline README.md:76]: Typo: `override`

File: udata_hydra/utils/json.py [hunk 3]
```diff
@@ -1,11 +0,0 @@
...
-
-
-def is_json_file(filename: str) -> bool:
-    """Checks if a string is a valid json"""
-    try:
-        with open(filename) as f:
-            json.load(f)
-            return True
-    except ValueError:
-        return False
```
Comment (magopian) [inline udata_hydra/utils/json.py:11]: Was that leftover code? Well spotted!
Do we have some kind of tooling in the python world to automatically detect leftover code?
Being a dynamic language, I'm not even sure this is possible to be honest 😬

**Review by magopian — APPROVED** [review]
Test coverage, nice!
Do you want to add this to the CHANGELOG?

Some future discussion maybe: do we want to enable a CI target to fail when a PR lowers the global test coverage?

**Comment by bolinocroustibat** [comment]
> Test coverage, nice! Do you want to add this to the CHANGELOG?

Indeed, fixed

> 
> Some future discussion maybe: do we want to enable a CI target to fail when a PR lowers the global test coverage?

That's a good idea, but let's not make the stack overkill though, and slowing us down for fixes and features. This should only be done if this not becomes a blocker when needing to push fixes quickly. We can create a ticket as a reminder!

**Pull request merged**